lambdas/services/update_upload_state_service.py

```diff
@@ -0,0 +1,79 @@
+import os
+from datetime import datetime, timezone
+
+from botocore.exceptions import ClientError
+from enums.lambda_error import LambdaError
+from enums.metadata_field_names import DocumentReferenceMetadataFields
+from enums.supported_document_types import SupportedDocumentTypes
+from services.base.dynamo_service import DynamoDBService
+from utils.audit_logging_setup import LoggingService
+from utils.lambda_exceptions import UpdateUploadStateException
+
+logger = LoggingService(__name__)
+DATE_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+FAILED_FIELDS_ERROR = "Validation error on files and fields"
+Fields = DocumentReferenceMetadataFields
+
+
+class UpdateUploadStateService:
+    def __init__(self):
+        self.dynamo_service = DynamoDBService()
+        self.lg_dynamo_table = os.getenv("LLOYD_GEORGE_DYNAMODB_NAME")
+        self.arf_dynamo_table = os.getenv("DOCUMENT_STORE_DYNAMODB_NAME")
+
+    def handle_update_state(self, event_body):
+        try:
+            files = event_body["files"]
+            for file in files:
+                doc_ref = file["reference"]
+                doc_type = file["type"]
+                uploaded = file["fields"][Fields.UPLOADING.value]
```

Comment:
What if `uploaded` is a random string instead of a boolean? Maybe you can have a check here.
`file["fields"][Fields.UPLOADING.value] == "true"`